### PR TITLE
NXP-25385: deal with null change token

### PR DESCRIPTION
**Pull request by efge**

### CODE REVIEW

**Comment by covolution** [comment]
:+1:

File: nuxeo-features/nuxeo-thumbnail/src/main/java/org/nuxeo/ecm/platform/thumbnail/io/ThumbnailJsonEnricher.java [hunk 1]
```diff
@@ -80,7 +81,7 @@ public void write(JsonGenerator jg, DocumentModel document) throws IOException {
         jg.writeStringField(THUMBNAIL_URL_LABEL,
                 String.format(THUMBNAIL_URL_PATTERN, ctx.getBaseUrl().replaceAll("/$", ""),
                         document.getRepositoryName(), document.getId(),
-                        URLEncoder.encode(document.getChangeToken(), "UTF-8")));
+                        URLEncoder.encode(defaultString(document.getChangeToken()), "UTF-8")));
```
Comment (bdelbosc) [inline nuxeo-features/nuxeo-thumbnail/src/main/java/org/nuxeo/ecm/platform/thumbnail/io/ThumbnailJsonEnricher.java]: commons-lang3 dependency must be added to the pom.xml

**Comment by guirenard** [comment]
I'd rather do not add `changeToken` param name to the url at all if empty. Goal is to avoid a url like

`@rendition/thumbnail?changeToken=&size=tiny` from being cache by https://github.com/nuxeo/nuxeo/blob/master/nuxeo-services/nuxeo-platform-web-common/src/main/resources/OSGI-INF/web-request-controller-contrib.xml#L47

```
diff --git i/nuxeo-features/nuxeo-thumbnail/src/main/java/org/nuxeo/ecm/platform/thumbnail/io/ThumbnailJsonEnricher.java w/nuxeo-features/nuxeo-thumbnail/src/main/java/org/nuxeo/ecm/platform/thumbnail/io/Thumbna
index 19f417570dc..11fafd1cea3 100644
--- i/nuxeo-features/nuxeo-thumbnail/src/main/java/org/nuxeo/ecm/platform/thumbnail/io/ThumbnailJsonEnricher.java
+++ w/nuxeo-features/nuxeo-thumbnail/src/main/java/org/nuxeo/ecm/platform/thumbnail/io/ThumbnailJsonEnricher.java
@@ -66,8 +66,7 @@ public class ThumbnailJsonEnricher extends AbstractJsonEnricher<DocumentModel> {
 
     public static final String THUMBNAIL_URL_LABEL = "url";
 
-    public static final String THUMBNAIL_URL_PATTERN = "%s/api/v1/repo/%s/id/%s/@rendition/thumbnail?"
-            + CoreSession.CHANGE_TOKEN + "=%s";
+    public static final String THUMBNAIL_URL_PATTERN = "%s/api/v1/repo/%s/id/%s/@rendition/thumbnail?";
 
     public ThumbnailJsonEnricher() {
         super(NAME);
@@ -77,10 +76,13 @@ public class ThumbnailJsonEnricher extends AbstractJsonEnricher<DocumentModel> {
     public void write(JsonGenerator jg, DocumentModel document) throws IOException {
         jg.writeFieldName(NAME);
         jg.writeStartObject();
-        jg.writeStringField(THUMBNAIL_URL_LABEL,
-                String.format(THUMBNAIL_URL_PATTERN, ctx.getBaseUrl().replaceAll("/$", ""),
-                        document.getRepositoryName(), document.getId(),
-                        URLEncoder.encode(document.getChangeToken(), "UTF-8")));
+        String url = String.format(THUMBNAIL_URL_PATTERN, ctx.getBaseUrl().replaceAll("/$", ""),
+                document.getRepositoryName(), document.getId());
+        String ct = document.getChangeToken();
+        if (ct != null) {
+            url += CoreSession.CHANGE_TOKEN + URLEncoder.encode(document.getChangeToken(), "UTF-8");
+        }
+        jg.writeStringField(THUMBNAIL_URL_LABEL, url);
         jg.writeEndObject();
 
     }
```

File: nuxeo-features/nuxeo-thumbnail/src/main/java/org/nuxeo/ecm/platform/thumbnail/io/ThumbnailJsonEnricher.java (hunk 1 above)
Comment (efge) [reply nuxeo-features/nuxeo-thumbnail/src/main/java/org/nuxeo/ecm/platform/thumbnail/io/ThumbnailJsonEnricher.java]: No need, it's an indirect dependency already

**Comment by efge** [comment]
Let's clean up for 10.10 after this is cherry-picked in 10.3

**Pull request merged**